addon/wait-for-promise.ts

```diff
@@ -1,14 +1,20 @@
 import { DEBUG } from '@glimmer/env';
-import { Promise } from 'rsvp';
+import RSVP from 'rsvp';
 import buildWaiter from './build-waiter';
 
 const PROMISE_WAITER = buildWaiter('promise-waiter');
 
+type PromiseType<T> = Promise<T> | RSVP.Promise<T>;
+
+interface Thenable<T, Return extends PromiseType<T>> {
```

Comment:
Why did we need to add this? Also, can we not use `PromiseLike<T>` in this case, as it's basically a thenable interface?

Reply:
`PromiseLike<T>` doesn't have `catch`
![image](https://user-images.githubusercontent.com/199018/78461468-2c160000-7697-11ea-818c-6d6b4f264367.png)

`Thenable` doesn't either, but it doesn't need to, because `PromiseType` does have `catch`. :shrug:

`PromiseLike<T>` also for some reason isn't compatible with `PromiseType` like `Thenable` is. :-\

idk. I don't like it, but I don't know a better way, and TS isn't mature enough yet to lead me to a better way :(

Reply:
Ya I sorta knew this was the reason, as your `Thenable` implementation allows you to specify the `T` (value) _and_ `Return` (promise type, whether `Promise` or `RSVP.Promise`), whereas `PromiseLike` doesn't allow you to do that.

It's fuggly (as you said), but does work. It's a bit irritating to have to jump through these sorts of hooks, but that's the cost of typing I guess.

Reply:
Def the cost of typing when we're only sprinkling types on top of untyped stuff. 
if stuff were typed to begin with, everything would be much smoother :D

Reply:
> if stuff were typed to begin with, everything would be much smoother

agree to disagree :smile:

Reply:
I don't think that's a disagreement? :) Idk
It's more 'in addition to' I guess.

Like, in natively typed languages i've had a much easier time resolving type errors.